azurerm/resource_arm_virtual_machine_scale_set_test.go

```diff
@@ -437,6 +437,25 @@ func TestAccAzureRMVirtualMachineScaleSet_overprovision(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMVirtualMachineScaleSet_priority(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMVirtualMachineScaleSetPriorityTemplate(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineScaleSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualMachineScaleSetExists("azurerm_virtual_machine_scale_set.test"),
+					testCheckAzureRMVirtualMachineScaleSetPriority("azurerm_virtual_machine_scale_set.test", "Low"),
```

Comment:
Unless there is a reason for us to query Azure directly could we change this to check the state? like:
```go
resource.TestCheckResourceAttr(resourceName, "azurerm_virtual_machine_scale_set.test. priority", "low"),
```

I know this isn't like many of the tests in the resource, but without good reason this is how we should be verifying the new property as it also checks the state.